### ISSUE TITLE
Error on company completion (LSP PHP intelephense)

### ISSUE THREAD

**Issue by gjm** [doomemacs/doomemacs#6930]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27 or 28. *(Doom does not support development builds like 29+ or any version ending in .50 or .9x)*

### Expected behavior

Given this snippet (| being the cursor position):

```php
class Foo
{
    |
}
```

If I start typing `pub` I expect the completion system to popup with (at least) the `public` alternative. 

### Current behavior

I get the error:

```
Company: backend (:separate company-capf company-yasnippet company-dabbrev) error "Wrong type argument: list-or-vector-p, #s(hash-table size 1 test equal rehash-size 1.5 rehash-threshold 0.8125 data ("items" [#s(hash-table size 4 test equal rehash-size 1.5 rehash-threshold 0.8125 data ("label" "public" "kind" 14 "textEdit" #s(hash-table size 2 test equal rehash-size 1.5 rehash-threshold 0.8125 data ("range" #s(hash-table size 2 test equal rehash-size 1.5 rehash-threshold 0.8125 data ("start" #s(hash-table size 2 test equal rehash-size 1.5 rehash-threshold 0.8125 data ("line" 16 "character" 4)) "end" #s(hash-table size 2 test equal rehash-size 1.5 rehash-threshold 0.8125 data ("line" 16 "character" 7)))) "newText" "public")) "sortText" "public"))]))" with args (candidates pub)
```

Backtrace:

```
Debugger entered--Lisp error: (wrong-type-argument list-or-vector-p #<hash-table equal 1/1 0x1aa3f59>)
  lsp-completion--sort-completions(#<hash-table equal 1/1 0x1aa3f59>)
  #f(compiled-function () #<bytecode 0x172de601b9756ae>)()
  #f(compiled-function (probe pred action) #<bytecode -0x1d1768f596f09ab4>)("" nil t)
  all-completions("" #f(compiled-function (probe pred action) #<bytecode -0x1d1768f596f09ab4>) nil)
  completion-pcm--all-completions("" (point) #f(compiled-function (probe pred action) #<bytecode -0x1d1768f596f09ab4>) nil)
  completion-basic-all-completions("" #f(compiled-function (probe pred action) #<bytecode -0x1d1768f596f09ab4>) nil 0)
  lsp-completion-passthrough-all-completions("pub" #f(compiled-function (probe pred action) #<bytecode -0x1d1768f596f09ab4>) nil 3)
  #f(compiled-function (style) #<bytecode 0x1186b7e458e8ad7b>)(lsp-passthrough)
  completion--some(#f(compiled-function (style) #<bytecode 0x1186b7e458e8ad7b>) (lsp-passthrough basic partial-completion orderless))
  completion--nth-completion(2 "pub" #f(compiled-function (probe pred action) #<bytecode -0x1d1768f596f09ab4>) nil 3 (metadata (category . lsp-capf) (display-sort-function . identity) (cycle-sort-function . identity)))
  #f(compiled-function (string table pred point &optional metadata) "List the possible completions of STRING in completion table TABLE.\nOnly the elements of table that satisfy predicate PRED are considered.\nPOINT is the position of point within STRING.\nThe return value is a list of completions and may contain the base-size\nin the last `cdr'." #<bytecode -0x187a3a4ed476277a>)("pub" #f(compiled-function (probe pred action) #<bytecode -0x1d1768f596f09ab4>) nil 3 (metadata (category . lsp-capf) (display-sort-function . identity) (cycle-sort-function . identity)))
  apply(#f(compiled-function (string table pred point &optional metadata) "List the possible completions of STRING in completion table TABLE.\nOnly the elements of table that satisfy predicate PRED are considered.\nPOINT is the position of point within STRING.\nThe return value is a list of completions and may contain the base-size\nin the last `cdr'." #<bytecode -0x187a3a4ed476277a>) ("pub" #f(compiled-function (probe pred action) #<bytecode -0x1d1768f596f09ab4>) nil 3 (metadata (category . lsp-capf) (display-sort-function . identity) (cycle-sort-function . identity))))
  completion-all-completions("pub" #f(compiled-function (probe pred action) #<bytecode -0x1d1768f596f09ab4>) nil 3 (metadata (category . lsp-capf) (display-sort-function . identity) (cycle-sort-function . identity)))
  #<subr company-capf--candidates>("pub")
  apply(#<subr company-capf--candidates> "pub")
  (let ((orderless-match-faces [completions-common-part]) (completion-styles +vertico-company-completion-styles)) (apply fn args))
  +vertico--company-capf--candidates-a(#<subr company-capf--candidates> "pub")
  apply(+vertico--company-capf--candidates-a #<subr company-capf--candidates> "pub")
  company-capf--candidates("pub")
  company-capf(candidates "pub")
  company--multi-backend-adapter-candidates((company-capf company-yasnippet company-dabbrev) "pub" (:separate company-capf company-yasnippet company-dabbrev))
  company--multi-backend-adapter((:separate company-capf company-yasnippet company-dabbrev) candidates "pub")
  apply(company--multi-backend-adapter (:separate company-capf company-yasnippet company-dabbrev) (candidates "pub"))
  company-call-backend-raw(candidates "pub")
  company--fetch-candidates("pub")
  company-calculate-candidates("pub" nil)
  company--begin-new()
  company--perform()
  company-auto-begin()
  company-idle-begin(#<buffer DataTransferObject.php> #<window 3 on DataTransferObject.php> 6390 357)
  apply(company-idle-begin (#<buffer DataTransferObject.php> #<window 3 on DataTransferObject.php> 6390 357))
  explain-pause--wrap-callback(#s(explain-pause-command-record :command timer :native t :parent #s(explain-pause-command-record :command company-post-command :native nil :parent #s(explain-pause-command-record :command post-command-hook :native nil :parent #s(explain-pause-command-record :command root-emacs :native nil :parent nil :executing-time 1004 :entry-snap (25442 40770 411006 429000) :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 0) :executing-time 0 :entry-snap nil :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 1) :executing-time 0 :entry-snap (25442 40770 404572 906000) :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 2) :executing-time 0 :entry-snap nil :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 3) company-idle-begin #<buffer DataTransferObject.php> #<window 3 on DataTransferObject.php> 6390 357)
  apply(explain-pause--wrap-callback (#s(explain-pause-command-record :command timer :native t :parent #s(explain-pause-command-record :command company-post-command :native nil :parent #s(explain-pause-command-record :command post-command-hook :native nil :parent #s(explain-pause-command-record :command root-emacs :native nil :parent nil :executing-time 1004 :entry-snap (25442 40770 411006 429000) :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 0) :executing-time 0 :entry-snap nil :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 1) :executing-time 0 :entry-snap (25442 40770 404572 906000) :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 2) :executing-time 0 :entry-snap nil :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 3) company-idle-begin #<buffer DataTransferObject.php> #<window 3 on DataTransferObject.php> 6390 357))
  timer-event-handler([t 25442 40770 414581 nil explain-pause--wrap-callback (#s(explain-pause-command-record :command timer :native t :parent #s(explain-pause-command-record :command company-post-command :native nil :parent #s(explain-pause-command-record :command post-command-hook :native nil :parent #s(explain-pause-command-record :command root-emacs :native nil :parent nil :executing-time 1004 :entry-snap (25442 40770 411006 429000) :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 0) :executing-time 0 :entry-snap nil :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 1) :executing-time 0 :entry-snap (25442 40770 404572 906000) :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 2) :executing-time 0 :entry-snap nil :too-slow nil :is-profiled nil :under-profile nil :profile nil :depth 3) company-idle-begin #<buffer DataTransferObject.php> #<window 3 on DataTransferObject.php> 6390 357) nil 793000 nil])
```

### Steps to reproduce

1. Open Emacs
2. Open a PHP file with LSP and intelephense
3. Do as described in **Expected behavior**

### System Information

https://pastebin.com/5LnQc9x9

**Comment by enotskellington** [comment]
I had the same issue, after unpinning both `company` and `yasnippet` everything started working correctly.

node version: 19.1.0
emacs version: 28.1
doom version: 3.0.0-pre

I'm not sure what version of modules but I ran `sync -u` today.

**Comment by gjm** [comment]
@enotskellington Haven't had any issues unpinning? I'm afraid not using doom's pinned package version can bring unexpected behavior.

You use `unpin!` to unpin packages? If I encounter any issues, removing the `unpin!` and `doom sync` will rever to the pinned version?

**Comment by enotskellington** [comment]
I haven't seen any issues yet, but this is still a debugging step. I think after confirming no other packages are affected we cold propose an upgrade for these packages. (I'm not sure how to check which commit is being used when unpinning)

I fist tried just `yasnippet` as that was part of the error message but the issue persisted, so then also tried `company` which includes a number of dependencies. It's hard to say what exactly fixed the issue, but so far it is working well.

You can always remove any `unpin!` statements from your `.doom.d/packages.el` and then sync again to revert the change. Even if that didn't work, clearing out `.emacs.d/` and installing doom again will start fresh but preserve any customization in `.doom.d/`

**Comment by gjm** [comment]
Well I tried it but had no luck. Tried unpinning both packages and just each one of them, but no luck...

Thanks for your time and help.

**Comment by enotskellington** [comment]
Did you run `doom sync -u`? That will update the package after unpinning.

Jan 28, 2023 12:30:20 PM Gonçalo Marrafa ***@***.***>:

**Comment by enotskellington** [comment]
Just for reference, the package commits that I have working are:

- company-mode: `e5c7432607d8978ec10942c99ba636e7e2cd79df`
- yasnippet: `93b0284e5a6be8506d194aab00fb613be0f364fd`

**Comment by gjm** [comment]
I did `doom sync -u` after the changes. Event tried removing the whole `straight`  dir with no luck. 

How do you know which commit is being used? I `cd`'d into `straight/repos/company-mode` , did a `git log` and got
```
commit 1005540b1cdf176cbcf893b2fa83d2075cbbe3ca (grafted, HEAD -> master)
Author: Dmitry Gutov <dgutov@yandex.ru>
Date:   Thu Apr 7 02:23:44 2022 +0300

    Merge pull request #1320 from ilohmar/capf-tags
    
    company-capf: only rm tags-capf from capf if we use company-etags
```

Almost a year old! That can't be correct can it?

**Comment by enotskellington** [comment]
no that is not correct. I get:
```
commit e5c7432607d8978ec10942c99ba636e7e2cd79df (HEAD -> master, origin/master)
Author: Dmitry Gutov <dgutov@yandex.ru>
Date:   Thu Jan 26 05:13:06 2023 +0200

    company--render-icons-margin: Use >= to compare font height to target size
    
    For the rare cases where default-font-height is exactly (* 2 base-size).
    
    Brought up in https://github.com/company-mode/company-mode/discussions/1364.
```

can you share your packages.el?

**Comment by gjm** [comment]
Here's my packages.el:

```
(package! smart-newline)
(package! blackout)
(package! fancy-narrow)
(package! visual-regexp)
(package! visual-regexp-steroids)
(package! smart-comment)
(package! banner-comment)
(package! vmd-mode)
(package! company-prescient)
(package! company-quickhelp)
(package! benchmark-init)
(package! modus-themes)
(package! ef-themes)
(package! standard-themes)
(package! posframe)

(package! solaire-mode :disable t)

;; And your suggestion
(unpin! company-mode)
```

It's very strange to me that, even after clearing the repos, it still checks out an old commit...

**Comment by enotskellington** [comment]
It's just `(unpin! company)`. I did also unpin `yasnippet` but I think it was company that fixed the issue for me.

If the issue persists even with the latest version, I would try disabling your other company packages one at a time to see if they conflict.

Jan 30, 2023 1:55:37 AM Gonçalo Marrafa ***@***.***>:

**Comment by gjm** [comment]
Did what you suggested. Still no luck... Same result. Now, at least, I can confirm company is actually up-to-date

```
Author: Dmitry Gutov <dgutov@yandex.ru>
Date:   Sun Jan 29 21:29:55 2023 +0200

    Merge pull request #1366 from yugaego/maint
    
    Bits of Maintenance
```

Have no clue what can be wrong... 

Thanks for your help, once again.

**Comment by enotskellington** [comment]
Maybe both company and yasnippet? But otherwise I'm not sure what did it.

Jan 30, 2023 12:02:52 PM Gonçalo Marrafa ***@***.***>:

**Comment by gjm** [comment]
I did `unpin!` both. Maybe some other config issue... Can you share your `packages.el`?

**Comment by enotskellington** [comment]
```

(package! arduino-mode)
(unpin! yasnippet company)
```

maybe try removing all the other packages temporarily? If that works then you might have to add them back on at a time to see which conflicts.

**Comment by enotskellington** [comment]
Then in `init.el` I have the following relevant lines:
```
:tools
(lsp)

:lang
(php +lsp +tree-sitter)
```
And have installed intelephense globally with npm.

**Comment by enotskellington** [comment]
I have tested with `company-quickhelp` and `company-prescience` with success.

At one point I thought installing quickhelp broke it, so I nuked the whole thing and reinstalled doom. It still seemed to be broken, all I did was install intelephense globally again, which makes no sense as it was already installed, and it is working. Or maybe I had to wait for some task to complete in the background that I was unaware of. Perhaps try opening a php file and just waiting five minutes? After this it works immediately after opening the file, quick help pops up nice and fast. Very nice experience once it works.
